app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+class QuestionsController < ApplicationController
+  before_action :authenticate_user!, only: %i[create update destroy edit]
+  before_action :authorize_moderator, only: %i[create update destroy]
+  before_action :set_question, only: %i[show edit update destroy]
+  before_action :check_question_bank, only: %i[index show edit update destroy]
+
+  def index
+    @questions = Question.all
+    @questions = Question.paginate(page: params[:page], per_page: 6)
```

Comment:
We don't need to load all the questions here.

```suggestion
    @questions = Question.paginate(page: params[:page], per_page: 6)
```